src/Fields/field.jl

```diff
@@ -245,6 +245,10 @@ function offset_windowed_data(data, Loc, grid, indices)
         windowed_parent = parent(data)
     else
         parent_indices = map(parent_index_range, indices, loc, topo, halo)
+        if size(data)[3] == 1 && first.(axes(data))[3] == grid.Nz + 1 # take ssh into consideration
+            parent_indices = collect(parent_indices)
+            parent_indices[3] = 1:1
+        end
```

Comment:
this seems like a hack

also, what does `ssh` refers to?

Reply:
Yes, it is a temporary hack. I have outlined the issue arising in the absence of this hack in issue #3572 and created PR #3573 to resolve it. In the comment, 'ssh' refers to variables like shh, for which the parent array takes the z-halo into consideration even though it does not exist.

Reply:
OK. Well we won't merge a hack, right?

Reply:
No, once I fix it in PR #3573, we will import the fix (not hack) in this PR.

Reply:
OK, I see. This PR requires first #3573. Gotcha!

Reply:
I don't understand though why filling halos in z for free surface has a problem now; that was OK up to now, right? Why is #3573 holding us here?

Reply:
@siddharthabishnu could you merge `main`? I think then this comment is not relevant any more and can be resolved ;)